torchvision/models/densenet.py

```diff
@@ -69,11 +69,11 @@ def __init__(self, num_layers, num_input_features, bn_size, growth_rate, drop_ra
             self.add_module('denselayer%d' % (i + 1), layer)
 
     def forward(self, init_features):
-        features = [init_features]
+        features = init_features
         for name, layer in self.named_children():
-            new_features = layer(*features)
-            features.append(new_features)
-        return torch.cat(features, 1)
+            new_features = layer(features)
+            features = torch.cat([features, new_features], 1)
```

Comment:
I think with this the performance might take quite a hit: whereas before we only called `torch.cat()` once after all `features` were gathered, with this change we would call it for each item in `self.named_children()`. I think 

> seems more natural

is not a good reason here, if my feeling is right. If I'm wrong, feel free to ignore this.

Reply:
@pmeier 
Okay, I understand your comment!

Reply:
My intuition says yes, but I don't know for sure. I imagine that `torch.cat()` copies all values to contiguous memory locations, whereas this is not necessary for `list.append()`. Could you do timing analysis to determine whether your change doesn't affect the performance alot or my intuition is right?

Reply:
@pmeier 

I tested it. You are right. It is not efficient to call torch.cat() every iteration.

Thank you for your review!!

Reply:
@pmeier is right. appending in a list is very cheap, while calling `torch.cat` re-copies all the memory